### PR TITLE
feat(ui): Add links to wft, cwf, or cwft to workflow list and details. Closes #3621

### LINKED ISSUE #3621: Make it easier to find out the cron/template/cluster template that the workflow was submitted from
# Summary

Show this is the _workflow list_ and _workflow summary panel_.

# Motivation

As a user

* I'd like to quickly get from my workflow to the template that it was submitted from.
* List similar workflows.

# Proposal

UI only change.

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [ ] ~~I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.~~ N/A
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: ui/src/app/shared/components/resource-editor/resource-editor.tsx [hunk 1]
```diff
@@ -29,7 +29,11 @@ const LOCAL_STORAGE_KEY = 'ResourceEditorLang';
 
 export class ResourceEditor<T> extends React.Component<Props<T>, State> {
     private set lang(lang: string) {
-        this.setState({lang, value: stringify(parse(this.state.value), lang)});
+        try {
```
Comment (alexec) [inline ui/src/app/shared/components/resource-editor/resource-editor.tsx:32]: this bonus catches submission bugs

File: ui/src/app/workflows/components/workflow-drawer/workflow-drawer.tsx [hunk 2]
```diff
@@ -74,6 +75,11 @@ export class WorkflowDrawer extends React.Component<WorkflowDrawerProps, Workflo
                         </div>
                     </div>
                 )}
+                <div className='workflow-drawer__section'>
+                    <div className='workflow-drawer__workflowFrom'>
+                        <WorkflowFrom namespace={wf.metadata.namespace} labels={wf.metadata.labels} />
+                    </div>
+                </div>
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-drawer/workflow-drawer.tsx:82]: Screenshot 2020-08-05 at 10 10 51
Screenshot 2020-08-05 at 10 16 33

File: ui/src/app/workflows/components/workflow-summary-panel.tsx [hunk 3]
```diff
@@ -16,11 +20,25 @@ export const WorkflowSummaryPanel = (props: {workflow: Workflow}) => (
                 {title: 'Message', value: props.workflow.status.message},
                 {title: 'Name', value: props.workflow.metadata.name},
                 {title: 'Namespace', value: props.workflow.metadata.namespace},
+                {title: 'From', value: <WorkflowFrom namespace={props.workflow.metadata.namespace} labels={props.workflow.metadata.labels} />},
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-summary-panel.tsx:23]: Screenshot 2020-08-05 at 10 07 32
Screenshot 2020-08-05 at 10 07 21
Screenshot 2020-08-05 at 10 07 11

File: ui/src/app/cluster-workflow-templates/components/cluster-workflow-template-link.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import * as React from 'react';
+import {uiUrl} from '../../shared/base';
+import {LinkButton} from '../../shared/components/link-button';
+
+export const ClusterWorkflowTemplateLink = (props: {name: string}) => (
```
Comment (simster7) [inline ui/src/app/cluster-workflow-templates/components/cluster-workflow-template-link.tsx:5]: Why not generalize this component?

File: ui/src/app/workflows/components/workflow-drawer/workflow-drawer.tsx (hunk 2 above)
Comment (simster7) [reply ui/src/app/workflows/components/workflow-drawer/workflow-drawer.tsx:82]: I'm not sure I like this. A workflow can only be submitted from a single source (cwf, wftmpl, etc.) and this makes it seem like there could be more than one... almost as if it were labels or annotations. Can we make this a normal text field? `Submitted From: CronWorkflow omiscient-rhino` or something?

File: ui/src/app/workflows/components/workflow-from.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import {labels} from '../../../models';
+import {ClusterWorkflowTemplateLink} from '../../cluster-workflow-templates/components/cluster-workflow-template-link';
+import {CronWorkflowLink} from '../../cron-workflows/components/cron-workflow-link';
+import {WorkflowTemplateLink} from '../../workflow-templates/components/workflow-template-link';
+
+export const WorkflowFrom = (props: {namespace: string; labels: {[name: string]: string}}) => {
```
Comment (simster7) [inline ui/src/app/workflows/components/workflow-from.tsx:7]: Nice and compact

File: ui/src/app/workflows/components/workflow-drawer/workflow-drawer.tsx (hunk 2 above)
Comment (simster7) [reply ui/src/app/workflows/components/workflow-drawer/workflow-drawer.tsx:82]: Also, the icon is not entirely clean... I think we could benefit from spelling out `CronWorkflow`, etc.

File: ui/src/app/cluster-workflow-templates/components/cluster-workflow-template-link.tsx (hunk 4 above)
Comment (alexec) [reply ui/src/app/cluster-workflow-templates/components/cluster-workflow-template-link.tsx:5]: I can't do this for cluster workflows - I did expect to re-use this, but I've not done so could be inlined?

**Comment by alexec** [comment]
@simster7 I'm not sure what you mean by "Can we make this a normal text field?". A text field is not (typically) clickable, hence a button.  Do you want longer text?

**Comment by simster7** [comment]
Here is an example from the `CronWorkflow` status pane. This fields shows and links to all active Workflows from this Cron (the inverse of this PR). Just a suggestion

Where there are none running
image

When there is at least one running
image

**Comment by alexec** [comment]
# Cron Workflow Page
Screenshot 2020-08-05 at 14 55 42
# Workflow Drawer
Screenshot 2020-08-05 at 14 52 50
# Workflow Summary
Screenshot 2020-08-05 at 14 52 56

**Pull request merged**